crates/goose-cli/src/commands/term.rs

```diff
@@ -0,0 +1,243 @@
+use anyhow::{anyhow, Result};
+use goose::session::session_manager::SessionType;
+use goose::session::SessionManager;
+use uuid::Uuid;
+
+use crate::session::{build_session, SessionBuilderConfig};
+
+const TERMINAL_SESSION_PREFIX: &str = "term:";
+
+/// Ensure a terminal session exists, creating it if necessary
+async fn ensure_terminal_session(
+    session_name: String,
+    working_dir: std::path::PathBuf,
+) -> Result<()> {
+    if SessionManager::get_session(&session_name, false)
+        .await
+        .is_err()
+    {
+        let session = SessionManager::create_session_with_id(
+            session_name.clone(),
+            working_dir,
+            session_name.clone(),
+            SessionType::User,
+        )
+        .await?;
+
+        SessionManager::update_session(&session.id)
+            .user_provided_name(session_name)
+            .apply()
+            .await?;
+    }
+    Ok(())
+}
+
+/// Handle `goose term init <shell>` - print shell initialization script
+pub fn handle_term_init(shell: &str) -> Result<()> {
+    let terminal_id = Uuid::new_v4().to_string();
```

Comment:
actually you could just create the session here and then use the session id of that session as the terminal_id